File: 2017/day02_1.py
# pylint: disable=missing-module-docstring, missing-function-docstring


from typing import Any


def solution(content: list[str]) -> Any:
    return sum(
        max(row) - min(row)
        for row in [[int(x) for x in line.split()] for line in content]
    )


def main() -> None:
    with open("day02.dat", "r", encoding="utf8") as file:
        print(solution(file.readlines()))


if __name__ == "__main__":
    main()

File: 2017/day02-1.ts
function solution(content: string) {
  return content
    .split("\n")
    .map((line) => line.split("\t").map((item) => parseInt(item)))
    .map((line) => Math.max(...line) - Math.min(...line))
    .reduce((acc, cur) => acc + cur, 0);
}

Deno.readTextFile("day02.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
